### ISSUE TITLE
A way to fire 'buffering' state externally while downloading media file.

### ISSUE THREAD

**Issue by Zensonaton** [media-kit/media-kit#671]
I've made my own player class that uses media_kit's `Player`. Here is one of the methods that my class provides:
```dart
Future<void> openAndPlay(MyAudioClass audio) async {
  final File cachedFile = DefaultCacheManager().getSingleFile(
    audio.url,
    key: audio.key,
  );

  await _player.open(
    Media(
      cachedFile.path,
    ),
  );

  // Update Discord's Rich Presence, Windows SMTC, etc.
}
```
As you can see, the `openAndPlay` method tries to retrieve file from cache or download if it's not present.
However, there is an issue with my current approach: `Player` object (and therefore, UI) doesn't know that `openAndPlay` method is trying to download a file before passing to it. Due to that, I'm unable to show "buffering" state in the UI of my app.

My question is, "is there a way to fire a 'buffering' state without calling `Player.open`?"

Technically, I can implement my own solution like:
```dart
bool get buffering => somehowCheckIfFileIsDownloading() || _player.state.buffering;
```
but I don't think that this is a great solution to this problem.

**Comment by alexmercerind** [comment]
Please don't do this.

**Comment by alexmercerind** [comment]
TL;DR: This is never going to reliable.

Caching is WIP: #269 & https://github.com/alexmercerind/network_cache_manager.

A lack of time & motivation.

**Comment by Zensonaton** [comment]
Okay, so it's not a great idea, makes sense, thank you for pointing that out 👀

Now I wonder if there is a callback method that returns media bytes when data get fully loaded in `Player.open(...)` method.
If so, there is a possibility of doing something like this:
```dart
_player.open(Media(isInCache ? file.path : url));
_player.someMagicCallback((bytes) => saveInCache(...));
```
